api/v1_stats.inc

```diff
@@ -33,6 +33,54 @@ function api_v1_stats_site($method, $data, $query_params)
     return $stats;
 }
 
+//---------------------------------------------------------------------------
+// stats/site/projects
+// Produces output like
+// ```
+// {
+//    "F1.proj_avail": 11,
+//    "F2.proj_avail": 7,
+//    "F2.proj_unavail": 2,
+//    "F2.proj_waiting": 1,
+//    ...
+//    "P3.proj_unavail": 1,
+//    "project_delete": 46,
+//    "project_new": 224,
+//    "proj_post_first_available": 4,
+//    "proj_post_first_checked_out": 24,
+//    "proj_post_first_unavailable": 3,
+//    "proj_post_second_available": 4,
+//    "proj_post_second_checked_out": 2,
+//    "proj_submit_pgposted": 15,
+//    "SR": 5
+// }```
+function api_v1_stats_site_projects($method, $data, $query_params)
+{
+    $output = [];
+
+    $sql = "
+        SELECT state, COUNT(*)
+        FROM projects
+        GROUP BY state
+    ";
+    $result = DPDatabase::query($sql);
+    while([$project_state, $count] = mysqli_fetch_row($result)) {
+        $output[$project_state] = (int)$count;
+    }
+
+    $sql = "
```

Comment:
Aside: I hate the way SR is hacked into our code.

SR isn't a _state_ of any kind (although it is a `Stage`) and unlike all of the others returned from this endpoint, it isn't a project state and thus isn't included in `v1/projects/states`. And although this endpoint doesn't _say_ "states" anywhere, it feels like we're mushing stuff together here.

Given that, what do you think about changing this endpoint to `v1/stats/projects/states` (which should include return values for all of the possible project states, with 0 for those without any current entries) and creating an additional endpoint `v1/stats/projects/stages` that groups all the projects into the stage (round, pool, or SR) and return the counts that way?

eg (but with actual counts):
```json
{
    "P1": 1,
    "P2": 2,
    "P3": 3,
    "F1": 4,
    "F2": 5,
    "PP": 7,
    "PV": 8,
    "SR": 9
}
```

Reply:
Thanks. I've failed to make the distinction between Stages and States multiple times before, so the reminder is helpful. I'll do as you suggested.

Reply:
@cpeel , are you intentionally lumping Unavailable, Bad, Available and Waiting together for the rounds, and similar for PP, or was that just an example?

Reply:
I am, yes. At this level they're all part of that "stage". If the user really cares about the nuanced view of available/waiting/etc they can process the project states and get whatever combination of data they are looking for.

Reply:
OK, done, but see caveat in commit message.